### PR TITLE
implement nD skimage.filters.farid (Farid & Simoncelli filter)

### LINKED ISSUE #5089: filters.farid should be generalised to n-dimensions
## Description

In #4347 we generalised Sobel, Prewitt, and Scharr filters to n dimensions. I think the Farid & Simoncelli filter, which has given me very good results in 2D, can also be generalised to nD, so we should do that.

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #5089

#4347 generalised Sobel, Prewitt, and Scharr filters to n dimensions.  This PR does the same for the Farid & Simoncelli case.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/filters/edges.py [hunk 1]
```diff
@@ -706,27 +712,41 @@ def laplace(image, ksize=3, mask=None):
     return _mask_filter_result(result, mask)
 
 
-def farid(image, *, mask=None):
-    """Find the edge magnitude using the Farid transform.
+def farid(image, mask=None, *, axis=None, mode='reflect', cval=0.0):
+    """Find the edge magnitude using the Scharr transform.
```
Comment (jni) [inline skimage/filters/edges.py:716]:
```suggestion
    """Find the edge magnitude using the Farid transform.
```

I guess a copy-paste gone astray?

File: skimage/filters/edges.py [hunk 2]
```diff
@@ -706,27 +712,41 @@ def laplace(image, ksize=3, mask=None):
...
+    cval : float, optional
+        When `mode` is ``'constant'``, this is the constant used in values
+        outside the boundary of the image data.
 
     Returns
     -------
-    output : 2-D array
-        The Farid edge map.
+    output : array of float
+        The Scharr edge map.
```
Comment (jni) [inline skimage/filters/edges.py:744]:
```suggestion
        The Farid edge map.
```

**Review by jni — APPROVED** [review]
Failures in pre-release are caused by networkx 2.7 deprecation warnings about 3.0, and should be fixed in a separate PR. Thanks @grlee77, this is great!

**Comment by grlee77** [comment]
Thanks for the review, I merged the suggestions

**Review by rfezzani — APPROVED** [review]
:+1: Thank you @grlee77, merging!

**Pull request merged**